x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -180,11 +180,12 @@ const getExpressionFromDocument = async (
   };
 };
 
-function getViewUnderlyingDataArgs({
+async function getViewUnderlyingDataArgs({
```

Comment:
There's no need to make this async and use the service.
`indexPatternsCache` already contains all used dataviews in the visualization, see here: https://github.com/elastic/kibana/pull/147000/files#diff-06b33562b1971e1caf72a3a592b2ecd93a4b7199fe9d1bf66f36bc9d906325c1R922